pkg/controller/dataimportcron-controller.go

```diff
@@ -1036,6 +1036,27 @@ func addDataImportCronControllerWatches(mgr manager.Manager, c controller.Contro
 		return err
 	}
 
+	//FIXME: nicer cron name parsing
+	mapCronJobToCron := func(_ context.Context, obj client.Object) []reconcile.Request {
+		cronLabel := obj.GetLabels()[common.DataImportCronLabel]
+		cronNsName := strings.Split(cronLabel, ".")
+		if len(cronNsName) != 2 || len(cronNsName[0]) == 0 || len(cronNsName[1]) == 0 {
```

Comment:
Are `len(cronNsName[0]) == 0 || len(cronNsName[1]) == 0` actually possible?

Reply:
Not really, unless somebody annotated the cronjob with "." or something, but we can ignore such case.
Note the whole conversion is buggy as already commented.